zui/src/setup.test.ts

```diff
@@ -0,0 +1,49 @@
+import { expect } from 'vitest'
+import { Project, Diagnostic } from 'ts-morph'
+
+export function isValidTypescript(
+  code: string,
+): { isValid: true } | { isValid: false; diagnostics: Diagnostic[]; errorMessage: string } {
+  const project = new Project({})
+  try {
+    project.createSourceFile('test.ts', code)
+    const diags = project.getPreEmitDiagnostics()
+    if (diags.length) {
+      return { isValid: false, diagnostics: diags, errorMessage: project.formatDiagnosticsWithColorAndContext(diags) }
+    }
+    return { isValid: true }
+  } catch (e: any) {
+    return { isValid: false, diagnostics: [], errorMessage: e?.message || '' }
+  }
+}
+
+expect.extend({
```

Comment:
**praise:** this is nice; are these functions properly typed outside this file ?

Reply:
yes, in `vitest.d.ts` the declarations extend the expect type to add the types of these two functions